scripts/changelog-pr.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+set -eu
+set -o pipefail
+
+if [ -z ${1+x} ]; then
+    >&2 echo "Error: pass version as argument. For example: $0 1.42.0 [release branch or main by default]"
+    exit
+fi
+
+VERSION=$1
+# Branch from which the release should be made
+if [ -z ${2+x} ]; then
+    RELEASE_BRANCH=main
+else
+    RELEASE_BRANCH=$2
+fi
+
+SEMVER_REGEX="^(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(\\-[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?(\\+[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?$"
+if [[ ! $VERSION =~ $SEMVER_REGEX ]] ; then
+    >&2 echo "Error: version $VERSION is not SemVer compatible(https://semver.org/)"
+    exit
+fi
+major=${BASH_REMATCH[1]}
+minor=${BASH_REMATCH[2]}
+
+if ! command -v gh &> /dev/null
+then
+    >&2 echo "Error: gh could not be found. See: https://cli.github.com/manual/"
+    exit
+fi
+
+if [[ "$(git branch --show-current)" != "${RELEASE_BRANCH}" ]]; then
+    >&2 echo "Error: release should be performed from ${RELEASE_BRANCH} branch"
+    exit
+fi
+
+if [[ "$(git diff)" != "" ]]; then
+    >&2 echo "Error: release should be performed from CLEAN (!!!) ${RELEASE_BRANCH} branch"
+    exit
+fi
+
+# add changelog
+cat > RELEASENOTES.md <<- EOM
+## Changelog
+EOM
+if [[ $major == "2" ]]; then
+  if [[ $minor == "0" ]]; then
+      echo "This is a patch release on top of v2.0.x which does not include changes introduced in v2.1.x and later.  " >> RELEASENOTES.md
+  fi
+  echo "For details on how v2 differs from v1 and guidance on migrating from v1 of the Mapbox Navigation SDK for Android to the v2 public preview, see 2.0 Navigation SDK Migration Guide.  " >> RELEASENOTES.md
+fi
+
+echo "For details on how v2 differs from v1 and guidance on migrating from v1 of the Mapbox Navigation SDK for Android to the v2 public preview, see 2.0 Navigation SDK Migration Guide." >> RELEASENOTES.md
+
+scripts/changelog.js --compile >> RELEASENOTES.md
+
+# remove unreleased changelogs
+rm -rf changelogs/unreleased
+
+cat >> RELEASENOTES.md <<- EOM
+### Mapbox dependencies
+This release depends on, and has been tested with, the following Mapbox dependencies:
+
+EOM
+cat ./build/dependencies.md >> RELEASENOTES.md
+
+# change CHANGELOG.md
+
+#TMPDIR=$(mktemp -d)
+#trap "rm -rf $TMPDIR" EXIT
+
+## extract header from old changelog
+#sed '/# Changelog/q' CHANGELOG.md > $TMPDIR/HEADER
+## extract old changelog entries
+#sed -n '/# Changelog/,$p' CHANGELOG.md | tail -n +2 > $TMPDIR/CURRENTCHANGELOG
+#
+## update changelog
+#cat $TMPDIR/HEADER > CHANGELOG.md
+#echo "" >> CHANGELOG.md
+#echo "## [$VERSION] - $(date '+%Y-%m-%d')" >> CHANGELOG.md
+#echo "" >> CHANGELOG.md
+#cat RELEASENOTES.md >> CHANGELOG.md
+#cat $TMPDIR/CURRENTCHANGELOG >> CHANGELOG.md
+
+# create release branch
```

Comment:
I noticed that libraries like [Kotlin Coroutines](https://github.com/Kotlin/kotlinx.coroutines) and [mapbox maps](https://github.com/mapbox/mapbox-maps-android) support both github releases and changelog file. So maybe we should do the same just because it's a standard 😄

Reply:
I will add support of updating `CHANGELOG.md`